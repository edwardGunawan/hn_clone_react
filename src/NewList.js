import React from 'react';
import PropTypes from 'prop-types';
const TimeAgo = require('react-timeago').default;
const parse = require('url-parse');


const NewList = ({by, descendants, time, title, type, url,score}) => {
    const url_parse = parse(url);
    return (
        <div>
            <h6>{title}<span>({url_parse.hostname})</span></h6>
            <p> {score} points by {by} <TimeAgo date={time*1000}/> | <span>
            {descendants} {(descendants === 0) ? 'discuss' : 'comments' }</span></p>
        </div>
    )
}

NewList.propTypes = {
    by: PropTypes.string.isRequired,
    descendants: PropTypes.number.isRequired,
    time: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
}
export default NewList;