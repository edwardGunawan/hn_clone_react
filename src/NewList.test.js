import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { NewList } from './NewList';

configure({ adapter: new Adapter() });

const oneNews = {
    by: "irsina",
    descendants: 17,
    id: 18265074,
    kids: [
        18265466,
        18265435,
        18265196,
        18265226,
        18265502,
        18265313
    ],
    score: 102,
    time: 1540065131,
    title: "Show HN: Lipreading with Deep Learning",
    type: "story",
    url: "https://github.com/astorfi/lip-reading-deeplearning"
}

describe('NewList Component', () => {
    let wrapper;
    const { by, descendants, kids, time, title, type, url, score,id } = oneNews;
    beforeEach(() => {
        wrapper = shallow(<NewList by={by}
            descendants={descendants}
            kids={kids}
            time={time}
            title={title}
            id={id}
            type={type}
            score={score}
            url={url} />);
    });

    it('should render <div/>', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render all props value', () => {
        expect(wrapper.text()).toEqual(
            `Show HN: Lipreading with Deep Learning(github.com) 102 points by irsina <TimeAgo /> | <Link />`
            );
    });

    // matching snapshot
    // it('should render correctly', () => {
    //     expect(wrapper).toMatchSnapshot();
    // });
})
