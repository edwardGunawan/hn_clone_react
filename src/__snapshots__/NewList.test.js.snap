// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewList Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewList
    by="irsina"
    descendants={17}
    kids={
      Array [
        18265466,
        18265435,
        18265196,
        18265226,
        18265502,
        18265313,
      ]
    }
    score={102}
    time={1540065131}
    title="Show HN: Lipreading with Deep Learning"
    type="story"
    url="https://github.com/astorfi/lip-reading-deeplearning"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h6>
          Show HN: Lipreading with Deep Learning
          <span>
            (
            github.com
            )
          </span>
        </h6>,
        <p>
           
          102
           points by 
          irsina
           
          <TimeAgo
            component="time"
            date={1540065131000}
            formatter={[Function]}
            live={true}
            maxPeriod={Infinity}
            minPeriod={0}
            now={[Function]}
          />
           | 
          <span>
            17
             
            comments
          </span>
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Show HN: Lipreading with Deep Learning",
            <span>
              (
              github.com
              )
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Show HN: Lipreading with Deep Learning",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "(",
                "github.com",
                ")",
              ],
            },
            "ref": null,
            "rendered": Array [
              "(",
              "github.com",
              ")",
            ],
            "type": "span",
          },
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            102,
            " points by ",
            "irsina",
            " ",
            <TimeAgo
              component="time"
              date={1540065131000}
              formatter={[Function]}
              live={true}
              maxPeriod={Infinity}
              minPeriod={0}
              now={[Function]}
            />,
            " | ",
            <span>
              17
               
              comments
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          102,
          " points by ",
          "irsina",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": "time",
              "date": 1540065131000,
              "formatter": [Function],
              "live": true,
              "maxPeriod": Infinity,
              "minPeriod": 0,
              "now": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " | ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                17,
                " ",
                "comments",
              ],
            },
            "ref": null,
            "rendered": Array [
              17,
              " ",
              "comments",
            ],
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h6>
            Show HN: Lipreading with Deep Learning
            <span>
              (
              github.com
              )
            </span>
          </h6>,
          <p>
             
            102
             points by 
            irsina
             
            <TimeAgo
              component="time"
              date={1540065131000}
              formatter={[Function]}
              live={true}
              maxPeriod={Infinity}
              minPeriod={0}
              now={[Function]}
            />
             | 
            <span>
              17
               
              comments
            </span>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Show HN: Lipreading with Deep Learning",
              <span>
                (
                github.com
                )
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Show HN: Lipreading with Deep Learning",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "(",
                  "github.com",
                  ")",
                ],
              },
              "ref": null,
              "rendered": Array [
                "(",
                "github.com",
                ")",
              ],
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              102,
              " points by ",
              "irsina",
              " ",
              <TimeAgo
                component="time"
                date={1540065131000}
                formatter={[Function]}
                live={true}
                maxPeriod={Infinity}
                minPeriod={0}
                now={[Function]}
              />,
              " | ",
              <span>
                17
                 
                comments
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            102,
            " points by ",
            "irsina",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": "time",
                "date": 1540065131000,
                "formatter": [Function],
                "live": true,
                "maxPeriod": Infinity,
                "minPeriod": 0,
                "now": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " | ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  17,
                  " ",
                  "comments",
                ],
              },
              "ref": null,
              "rendered": Array [
                17,
                " ",
                "comments",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
